packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -297,15 +299,30 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
       const videoOptions = { localVideoStreams: this.localStream ? [this.localStream] : undefined };
 
       const isTeamsMeeting = !('groupId' in this.locator);
-      const call = isTeamsMeeting
-        ? this.callAgent.join(this.locator as TeamsMeetingLinkLocator, {
-            audioOptions,
-            videoOptions
-          })
-        : this.callAgent.join(this.locator as GroupCallLocator, {
-            audioOptions,
-            videoOptions
-          });
+      /* @conditional-compile-remove(rooms) */
+      const isRoomsCall = !('roomId' in this.locator);
+
+      let call;
```

Comment:
Code lines from 302-325 could be wrapped into a function, (maybe name it joinCall?)

Then we can do conditional compilation in the function then:
const call = joinCall(locator, {audioOptions, videoOptions});

Reply:
Sure. Moved to private function.